servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcFilters.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.grpc.api;
+
+import io.servicetalk.concurrent.TimeSource;
+import io.servicetalk.concurrent.api.AsyncContext;
+import io.servicetalk.http.api.HttpRequestMetaData;
+import io.servicetalk.http.api.StreamingHttpClientFilterFactory;
+import io.servicetalk.http.api.StreamingHttpServiceFilterFactory;
+import io.servicetalk.http.utils.TimeoutHttpRequesterFilter;
+import io.servicetalk.http.utils.TimeoutHttpServiceFilter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.function.BiFunction;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.grpc.internal.DeadlineUtils.GRPC_DEADLINE_KEY;
+import static io.servicetalk.grpc.internal.DeadlineUtils.readTimeoutHeader;
+import static io.servicetalk.utils.internal.DurationUtils.ensurePositive;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+/**
+ * Utility filters for gRPC.
+ */
+public final class GrpcFilters {
```

Comment:
this doesn't strictly need to be in `grpc.api` (could go in `grpc.netty`) but I put it here to reference from javadocs on the builder method for easier discovery.